### PR TITLE
Show Fraud Protection API errors and other errors to users.

### PR DESCRIPTION
**Pull request by chadsowald**
This PR swaps out the default exception handling for custom error handling.  For Fraud Protection API errors it shows both the request and response full details to the user (sans the bearer token).  For other errors, it shows the stack trace with a link to open a bub in this repo (if they want to).  Here's what this looks like in practice.

image

image

### CODE REVIEW

File: src/ApplicationCore/Exceptions/FraudProtectionApiException.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public HttpResponseMessage Response { get; }
+
+        public FraudProtectionApiException()
+        {
+        }
+
+        public FraudProtectionApiException(HttpResponseMessage response) : base($"Fraud Protection API exception occurred!")
+        {
+            Response = response;
+        }
```
Comment (satyamamit) [inline src/ApplicationCore/Exceptions/FraudProtectionApiException.cs:20]: it looks you have just used this one in below code. Do you need all other methods, as most of the sample app code is written by you may be you need more than anyone.

File: src/Infrastructure/Services/FraudProtectionService.cs [hunk 2]
```diff
@@ -76,25 +76,27 @@ public FraudProtectionService(IOptions<FraudProtectionSettings> settings, IToken
...
 
-        private async Task<T> Read<T>(HttpResponseMessage rawResponse) where T : new()
+        private async Task<T> Read<T>(HttpResponseMessage response) where T : new()
         {
-            rawResponse.EnsureSuccessStatusCode();
-            var rawContent = await rawResponse.Content.ReadAsStringAsync();
-            rawResponse.Dispose();
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new FraudProtectionApiException(response);
```
Comment (satyamamit) [inline src/Infrastructure/Services/FraudProtectionService.cs:93]: thanks for making this change.

File: src/Web/Middleware/ExceptionHandlingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+{
+    public static class ExceptionHandlingMiddleware
+    {
+        public static void ConfigureExceptionHandler(this IApplicationBuilder app)
+        {
+            app.UseExceptionHandler(errorApp =>
+            {
+                errorApp.Run(async context =>
+                {
+                    context.Response.StatusCode = 500;
```
Comment (satyamamit) [inline src/Web/Middleware/ExceptionHandlingMiddleware.cs:24]: why are we hard coding it to 500, is it 500 in context of sample App?

Comment (chadsowald) [reply src/Web/Middleware/ExceptionHandlingMiddleware.cs:24]: Yes, that's right.  Sample site encountered an unexpected error in either case.  In reality, users won't see this 500 unless they look in Fiddler or something anyways, but good question!

File: src/ApplicationCore/Exceptions/FraudProtectionApiException.cs (hunk 1 above)
Comment (chadsowald) [reply src/ApplicationCore/Exceptions/FraudProtectionApiException.cs:20]: These are just the standard Exception constructors.  I agree either way, but don't mind this extra code to match the typical exception constructors so I'll leave it.  The other custom exception we have in this same folder has these constructors as well.

File: src/ApplicationCore/Exceptions/FraudProtectionApiException.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
```
[the remark this answers is not in the document]
Comment (chadsowald) [reply src/ApplicationCore/Exceptions/FraudProtectionApiException.cs:2]: Yes, we're producing it under the MIT license :-)

**Pull request merged**